File: TypeScript/problem_6_reverse_number.ts
namespace Problem6{
    const promptSync = require("prompt-sync")();

    function reverse(num: number): number {
        let result = 0
        let temp = 0
        while(num > 0){
            temp = num % 10
            result = (result * 10) + temp
            num = Math.floor(num/10);
        }
        return result
    }

    const num: number = parseInt(promptSync("Enter a number: ")!);
    console.log(`Reverse of ${num} is: ${reverse(num)}`);
}

File: Python/problem_6_reverse_number.py
def reverse(num):
    result = 0
    temp = 0
    while(num > 0):
        temp = num % 10
        result = (result * 10) + temp
        num = num//10

    return result



num = int(input("Enter a number: "))
print(f"Reverse of {num} is: {reverse(num)}")